homework/index.js

```diff
@@ -30,15 +32,126 @@
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
+  function renderContributors(contributors, contributorsContainer) {
+    createAndAppend('p', contributorsContainer, {
+      text: 'Contributors',
+      class: 'contributor-header',
+    });
+    contributors.forEach(contributor => {
+      const contributorList = createAndAppend('ul', contributorsContainer, {
+        class: 'contributor-list',
+      });
+
+      const contributorListItem = createAndAppend('li', contributorList);
+
+      const anchorLink = createAndAppend('a', contributorListItem, {
+        href: contributor.html_url,
+        target: '_blank',
+        class: 'contributor-item',
+      });
+
+      createAndAppend('img', anchorLink, {
+        src: contributor.avatar_url,
+        alt: contributor.login,
+        class: 'image',
+      });
+
+      const contributorDetails = createAndAppend('div', anchorLink, {
+        class: 'contributor-data',
+      });
+      createAndAppend('div', contributorDetails, { text: contributor.login });
+      createAndAppend('div', contributorDetails, {
+        text: contributor.contributions,
+        class: 'contributor-badge',
+      });
+    });
+  }
+
+  function fetchAndRenderData(selectedRepo, repoContainer, contributorsContainer, root) {
+    repoContainer.innerHTML = '';
+    contributorsContainer.innerHTML = '';
+
+    const updatedAt = new Date(selectedRepo.updated_at);
+
+    const repoTitles = { description: 'Description: ', forks: 'Forks: ', updated_at: 'Updated: ' };
+
+    const table = createAndAppend('table', repoContainer, { class: 'table' });
+    const tbody = createAndAppend('tBody', table);
+
+    const firstRow = createAndAppend('tr', tbody);
+    const leftCell = createAndAppend('td', firstRow);
+    const rightCell = createAndAppend('td', firstRow);
+
+    createAndAppend('span', leftCell, { text: 'Repository: ', class: 'repo-child left-cell' });
+    createAndAppend('a', rightCell, {
+      text: `${selectedRepo.name}`,
```

Comment:
There is not much point to use a template literal here. That only makes sense if you want to combine fix pieces of text with variable pieces, which is not the case here. You can just use the string directly:

```js
text: selectedRepo.name,
```

The same holds true for a couple of other places in your code.